src/Microsoft.Azure.WebJobs.Host/Executors/FunctionInstance.cs

```diff
@@ -51,6 +52,18 @@ public IFunctionInvoker Invoker
             get { return _invoker; }
         }
 
+        public object JobInstance
+        {
+            get
+            {
+                if (_jobInstance == null)
+                {
+                    _jobInstance = Invoker.CreateInstance();
```

Comment:
>_jobInstance = Invoker.CreateInstance(); [](start = 20, length = 40)

Can we just create this upfront in the ctor?
1. Ensures no races. 
2. lets the field be read-only like the others?